src/spikeinterface/preprocessing/filter.py

```diff
@@ -291,14 +333,77 @@ def __init__(self, recording, freq=3000, q=30, margin_ms=5.0, dtype=None):
         self._kwargs = dict(recording=recording, freq=freq, q=q, margin_ms=margin_ms, dtype=dtype.str)
 
 
+class CausalFilter(FilterRecording):
```

Comment:
Why having this class if the FilterRecording already handle it with specific options ?

Reply:
I guess it is for the same reason highpass_filter and bandpass_filter exist. Although it is true that in causal filtering, the users might need to provide some extra parameters compared with these other subclasses...

Reply:
I agree with @samuelgarcia we don't need the class in this case, since the `CasualFilter` class is really just instantiating the `FilterRecording` class (the `fix_dtype` and `kwargs` will be handled by the `FilterRecording` class).

My suggestion is to:
- remove this class
- explicittly define the `causal_filter` function (see my comment)